### PR TITLE
[WIP] Update documentation

### LINKED ISSUE #75: Documentation
Document everything nicely.

### PR DESCRIPTION
**Pull request by fredrikekre**
Fix #75

Atleast this is a better starting point than before.

Not sure I configured the deploy stuff correctly.

Review plz @KristofferC

### CODE REVIEW

**Comment by KristofferC** [comment]
Big improvement! Merge when you feel it is ok and I will check out if things are getting deployed correctly and go through it again.

File: docs/src/index.md [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
 
-* Evaluating shape functions, derivatives of shape functions etc. for different finite element spaces and quadrature rules.
-* Evaluating
-*
-* Exporting to meshes to VTK.
+* Facilitate integration using different quadrature rules.
+* Define different finite element spaces
+* Evaluate shape functions, derivatives of shape functions etc. for the different finite element spaces and quadrature rules.
+* Evaluate functions and derivatives in the finite element space.
+* Generate simple grids, importing grids from common file formats.
```
Comment (fredrikekre) [inline docs/src/index.md]: Soon™

File: docs/src/index.md [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
+* Facilitate degree of freedom numbering and keeping track of different fields.
```
Comment (fredrikekre) [inline docs/src/index.md]: Can add something like future section

**Comment by fredrikekre** [comment]
Is it possible to use jldoctest in docstrings? Seems like tests run, but I need to place a DocTestSetup somewhere to make it work. But where?

**Comment by KristofferC** [comment]
Julia Base has that a lot in their docstrings. Not sure how it works with modules though. Is it complaining about not finding JuAFEM?

**Comment by fredrikekre** [comment]
Exactly

**Comment by fredrikekre** [comment]
This did the trick:

https://github.com/JuliaDocs/Documenter.jl/issues/118#issuecomment-229470688

**Comment by fredrikekre** [comment]
Merge and see what happens? :)

**Comment by KristofferC** [comment]
go for it

**Comment by KristofferC** [comment]
Vi borde nog ha doctests i contmechtensors också.

**Comment by fredrikekre** [comment]
Incoming

**Comment by fredrikekre** [comment]
Men lite svårt när vi `rand`:ar tensorer :p

**Comment by KristofferC** [comment]
`srand`

**Comment by fredrikekre** [comment]
not implemented

**Comment by KristofferC** [comment]
wut?

**Comment by KristofferC** [comment]
```
@meta
 DocTestSetup = quote
srand(1234)
end
```

**Comment by KristofferC** [comment]
merge this now plx?

**Comment by KristofferC** [comment]
Cool!

 "WARNING: deploying docs with `GITHUB_API_KEY` is deprecated. Please use the new method discussed in..."

Will fix.

**Comment by KristofferC** [comment]
Should be good now...

**Comment by fredrikekre** [comment]
Heh, trodde `srand` var en deterministisk variant på `rand`. Så jag körde `srand(Vec{3})`
Like a newb.

**Comment by KristofferC** [comment]
xDD

**Pull request merged**